app/controllers/brapi/v1/germplasm_controller.rb

```diff
@@ -0,0 +1,250 @@
+class Brapi::V1::GermplasmController < Brapi::BaseController
+
+  attr_accessor :request_params, :user
+
+  before_filter :authenticate_api_key!
+
+
+  rescue_from 'ActiveRecord::InvalidForeignKey' do |exception|
+    message = exception.message.split("\n").try(:second)
+    attribute = message ? message[14..-1].split(')')[0] : ''
+    render json: { errors: { attribute: attribute, message: message } }, status: 422
+  end
+
+  rescue_from ActionController::ParameterMissing do |exception|
+    render json: { errors: { attribute: exception.param, message: exception.message } }, status: 422
+  end
+
+
+
+
+ def search
```

Comment:
I like to visually separate different areas, like 'search' from
rescue_from ActionController::ParameterMissing do |exception|
...